### DISCUSSION TITLE
How to open a directory without making nvim-tree occupying the entire window

### DISCUSSION THREAD

**Discussion by Prince213** [nvim-tree/nvim-tree.lua#1794] [Q&A]
Hi,
When I open a directory, nvim-tree occupies the entire window, I'm wondering if there's a way to make it open a small portion of the window. This has been asked at https://www.reddit.com/r/neovim/comments/urc32o/nvimtree_start_up/, but with no answer.
Thanks in advance.

**Comment by alex-courtis** [comment]
Unfortunately nvim-tree occupies the whole of its window.

I sounds like you might want a solution where a new window is opened for nvim-tree. Perhaps a floating window might be useful `:help nvim-tree.view.float`

Is your use case on vim start or opening a directory in an existing session?

Note that an overhaul of the startup behaviour is incoming #1669

**Reply by Prince213** [reply]
Hi,

> I sounds like you might want a solution where a new window is opened for nvim-tree.

It's more like having nvim-tree on the one side and an empty buffer on the other side.

> Is your use case on vim start or opening a directory in an existing session?

It's on vim start.

**Reply by alex-courtis** [reply]
Some sort of `new_window` option for the proposed startup options might be needed.

Please comment in the issue #1669 and describe exactly the desired behaviour i.e.
When you want a new nvim-tree window to open: what flavour of buffers should result in a new window.

**Reply by davidarny** [reply]
@alex-courtis I went through [this](https://github.com/nvim-tree/nvim-tree.lua/wiki/Open-At-Startup) wiki and tried all variants, but can make a tree to the side

Currently it opens like this:
<img width="1120" alt="43262" src="https://github.com/nvim-tree/nvim-tree.lua/assets/17799810/68596f5a-659b-4c3b-a6e5-9dbb21f00c02">

And I believe me and @Prince213 want something like this
<img width="1120" alt="90249" src="https://github.com/nvim-tree/nvim-tree.lua/assets/17799810/3ea508af-921d-4770-be6e-e568f198148a">

**Reply by alex-courtis** [reply]
I'm able to get the desired result with a ["Clean Room" Replication](https://github.com/nvim-tree/nvim-tree.lua/wiki/Troubleshooting#clean-room-replication) and the very basic open recipe at the bottom:

```lua
      watcher = false,
    },
  },
})

local function open_nvim_tree()

  -- open the tree
  require("nvim-tree.api").tree.open()
end

vim.api.nvim_create_autocmd({ "VimEnter" }, { callback = open_nvim_tree })

return M
```

Please try that out to see if it's a tree problem or something in your plugin / startup that is causing the isssue.
